### PR TITLE
feat: system-synced dark mode

### PR DESCRIPTION
**Pull request by erickzhao**
This PR adds a new setting that syncs your Fiddle theme with your OS. It's toggled on by default and will alternate between the default dark and light themes depending on your OS setting.

_(Please test this on Windows because I haven't yet 😇 )_

screencast 2020-08-28 15-52-21

### CODE REVIEW

File: src/renderer/state.ts [hunk 1]
```diff
@@ -808,7 +834,8 @@ export class AppState {
       | null
       | boolean,
   ) {
-    if (value) {
+    // null or undefined
+    if (value != null) {
```
Comment (erickzhao) [inline src/renderer/state.ts:838]: This is the one unrelated change I added. Previously, non-null falsy values would never get saved to localStorage.

File: src/main/system-theme.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  nativeTheme.on('updated', () => {
+    maybeUpdateTheme();
+  });
+
+  ipcMainManager.on(IpcEvents.SET_THEME_SOURCE, (_event, source) => {
+    nativeTheme.themeSource = source;
+  });
+}
+
+export function maybeUpdateTheme() {
```
Comment (HashimotoYT) [inline src/main/system-theme.ts:17]: really need export? never used anywhere rather that file

File: src/renderer/components/settings-general-appearance.tsx [hunk 3]
```diff
@@ -96,6 +106,11 @@ export class AppearanceSettings extends React.Component<
       this.setState({ themes, selectedTheme });
     });
 
+    autorun(async () => {
```
Comment (HashimotoYT) [inline src/renderer/components/settings-general-appearance.tsx:109]: What's the difference here when use `autorun`? It's something that I never see before so yeah, that looks strange for me.

**Review by vhashimotoo — DISMISSED** [review]
Looks interesting some questions:

Should be this enabled by default from start? We use the default theme from start, and now we change to the system default theme. Maybe leave that disabled by default.

Not sure if this bug, but when you select light theme but system default is dark the first click to the checkbox doesn't change the theme to dark:

recording

File: src/renderer/state.ts (hunk 1 above)
Comment (felixrieseberg) [inline src/renderer/state.ts:838]:
```suggestion
    if (value !== null && value !== undefined) {
```

**Comment by erickzhao** [comment]
@felixrieseberg now that you mention it, CSS might be a better approach here because it wouldn't require any main process work or IPC, and might have a lower maintenance burden. I think I tunnelled into the Electron-specific solution really quickly, but I'll take a closer look.
